lhapdf.sh

```diff
@@ -3,12 +3,19 @@ version: v6.1.5
 source: https://github.com/alisw/LHAPDF
 tag: v6.1.5
 requires:
-  - yaml-cpp
+ - yaml-cpp
+ - boost
+ - "autotools:(slc5).*"
+force_rebuild: true
 ---
 #!/bin/sh
 # Does not work out of source.
 
 cd $SOURCEDIR
+
+# autoreconf if using our own build of autotools
+[[ -n ${AUTOTOOLS_ROOT} ]] && autoreconf -ivf
```

Comment:
Neither for commands after while, until, if or in tests, nor for commands in chain with && or || does bash -e exit if a command fails - except for the last one (here autoreconf). And this is what we want.

I had tested this with SLC 5 and Ubuntu.